joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -129,12 +133,34 @@ JointTrajectoryController::state_interface_configuration() const
       conf.names.push_back(joint_name + "/" + interface_type);
     }
   }
+  if (!params_.speed_scaling_state_interface_name.empty())
+  {
+    conf.names.push_back(params_.speed_scaling_state_interface_name);
+  }
   return conf;
 }
 
 controller_interface::return_type JointTrajectoryController::update(
```

Comment:
I'm curious about one thing in the update method, if we set the initial default value, and in the first iteration of the speed scaling from the state is different, should we set the initial default one still?

I'm just curious what should be the behaviour in this case. @fmauch what should be done in your opinion?.

Thinking back may be the state interface one should superset the initial default one?.. this can be handled in a different PR

Reply:
Is it doing that?. As far as I looked, i didn't see that. I'll take a look again

Reply:
The scaling factor cmd is always set to the initial value on the first loop from the parameters right?  On the contrary, it should be the state value automatically on the first loop right?. 

This we can recheck and open an issue for future reference

Reply:
Actually, the cmd interface only gets set from the subscriber.

The state interface is always read at the beginning of `update()`. That value will be used for the cycle. That means, the initial value doesn't have an effect as soon as a state interface for reading the speed scaling is configured.

Reply:
Ok